purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesFactory.kt

```diff
@@ -302,25 +303,42 @@ internal class PurchasesFactory(
         }
     }
 
-    private fun createPaywallEventsManager(
+    private fun createEventsManager(
         context: Context,
         identityManager: IdentityManager,
         eventsDispatcher: Dispatcher,
         backend: Backend,
-    ): PaywallEventsManager? {
+    ): EventsManager? {
         // RevenueCatUI is Android 24+ so it should always enter here when using RevenueCatUI.
         // Still, we check for Android N or newer since we use Streams which are 24+ and the main SDK supports
         // older versions.
         return if (isAndroidNOrNewer()) {
-            PaywallEventsManager(
-                EventsFileHelper(
+            EventsManager(
+                legacyEventsFileHelper = EventsFileHelper(
                     FileHelper(context),
-                    PaywallEventsManager.PAYWALL_EVENTS_FILE_PATH,
+                    EventsManager.PAYWALL_EVENTS_FILE_PATH,
+                    PaywallStoredEvent::toString,
                     PaywallStoredEvent::fromString,
                 ),
```

Comment:
One thing I've thought is that maybe its worth just declaring PaywallsEventManager here instead, since it's already tested.